android/player/src/main/java/com/intuit/playerui/android/ui/PlayerFragment.kt

```diff
@@ -175,11 +178,14 @@ public abstract class PlayerFragment : Fragment(), ManagedPlayerState.Listener {
      */
     protected open fun handleAssetUpdate(asset: RenderableAsset?, animateTransition: Boolean) {
         lifecycleScope.launch(if (asset is SuspendableAsset<*>) Dispatchers.Default else Dispatchers.Main) {
-            try {
-                renderIntoPlayerCanvas(asset, animateTransition)
-            } catch (exception: Exception) {
-                exception.printStackTrace()
-                playerViewModel.fail("Error rendering asset", exception)
+            whenStarted { // TODO: This'll go away when we can call a suspend version of this
+                try {
+                    renderIntoPlayerCanvas(asset, animateTransition)
+                } catch (exception: Exception) {
+                    if (exception is CancellationException) throw exception
+                    exception.printStackTrace()
+                    playerViewModel.fail("Error rendering asset", exception)
+                }
```

Comment:
why is a CancellationException treated differently?

Reply:
Good question, `CancellationException` occurs if the coroutine is cancelled - so, in the case that the `PlayerFragment` is destroyed, it'll cancel the `PlayerViewModel.state` collection and any jobs spawned from it. If we're in the middle of rendering an asset, it'll throw inside `handleAssetUpdate` and be caught in this `catch`. To ensure we don't hinder cancellation and treat this case as a `PlayerViewModel.fail`, we need to re-throw.